modules/apps/headless/headless-builder/headless-builder-impl/src/test/java/com/liferay/headless/builder/internal/application/endpoint/EndpointMatcherTest.java

```diff
@@ -30,26 +30,35 @@ public class EndpointMatcherTest {
 	@Test
 	public void testGetEndpoint() {
 		List<APIApplication.Endpoint> endpoints = Arrays.asList(
-			_getEndpoint("/path-name", false),
-			_getEndpoint("/path-name2", false),
-			_getEndpoint("/path/{parameter}", true));
+			_getEndpoint(Http.Method.GET, "/path-name", false),
+			_getEndpoint(Http.Method.POST, "/path-name", false),
+			_getEndpoint(Http.Method.GET, "/path-name2", false),
```

Comment:
It would be also interesting to validate what if we ask for a POST or GET endpoint with a path of the opposite method, for example:

POST /path-name3

and we ask for the GET /path-name3

We are not validating that case

Reply:
Does this make sense to you?

```java
Assert.assertEquals(
			null,
			endpointMatcher.getEndpoint(Http.Method.POST, "/path-name2", null));
```